silkworm/db/snapshots/index/bloom_filter.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+   Copyright 2024 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "bloom_filter.hpp"
+
+#include <numbers>
+#include <random>
+#include <sstream>
+#include <string_view>
+
+#include <openssl/evp.h>
+
+#include <silkworm/core/common/bytes_to_string.hpp>
+#include <silkworm/core/common/endian.hpp>
+#include <silkworm/core/common/util.hpp>
+
+namespace silkworm::snapshots::index {
+
+using namespace std::numbers;
+
+//! The minimum Bloom filter bits count
+static constexpr size_t kMinimumBitsCount{2};
+
+//! kRotation sets how much to rotate the hash on each filter iteration.
+//! This is somewhat randomly set to a prime on the lower segment of 64.
+constexpr size_t kRotation{17};
```

Comment:
I didn't mean to pick on code style, because this PR was done before the meeting, but just FYI after new code style updates, this will be changed to:

```suggestion
static constexpr std::size_t kRotation = 17;
```